### DISCUSSION TITLE
scrollIntoView method throwing error "JavascriptException: javascript error: missing ) after argument list"

### DISCUSSION THREAD

**Discussion by aravindraveendran7** [selenide/selenide#2403] [Q&A] [answered]
1. I have used 
```java
$(byXpath(element)).shouldBe(enabled).scrollTo().click()
```

But this code doesn't always scroll to the intended element. It works at times and sometimes not. What could be the solution for this?

2. Also, I have tried this:
```java
$(byXpath(formatedElement)).shouldBe(enabled).scrollTo().scrollIntoView(elementToBeSearched).click(usingJavaScript())
```
which gives me the error `JavascriptException: javascript error: missing ) after argument list`.

What is the best way to scroll to an element which we can rely on always?

**Comment by asolntsev** [comment]
Hi @aravindraveendran7 .
What is `elementToBeSearched` in your code sample? 

Method `$. scrollIntoView(...)` expected either `boolean` or `String` parameter. 
If it's String, it expects so-called "scroll options", not an element. Read the javadoc:

```java
  /**
   * Ask browser to scroll the element on which it's called into the visible area of the browser window.
   * <pre>
   * scrollIntoViewOptions:
   *  * behavior (optional) - Defines the transition animation
   *    1. auto (default)
   *    2. instant
   *    3. smooth
   *  * block (optional)
   *    1. start (default)
   *    2. center
   *    3. end
   *    4. nearest
   *  * inline
   *    1. start
   *    2. center
   *    3. end
   *    4. nearest (default)
   * </pre>
   *
   * <b>Usage:</b>
   * <pre>
   *   element.scrollIntoView("{block: \"end\"}");
   *   element.scrollIntoView("{behavior: \"instant\", block: \"end\", inline: \"nearest\"}");
   * </pre>
   *
   * @param scrollIntoViewOptions is an object with the align properties: behavior, block and inline.
   * @see com.codeborne.selenide.commands.ScrollIntoView
   * @see <a href="https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView">Web API reference</a>
   */
  @Nonnull
  @CanIgnoreReturnValue
  SelenideElement scrollIntoView(String scrollIntoViewOptions);
```

**Reply by aravindraveendran7** [reply]
@asolntsev Thank you for your reply. Here `elementToBeSearched` refers to the text element I need to scroll my view into. As this was expecting a String parameter, I assumed it needs to be a text from my application which it would scroll into.
Could you also please let me know why the first method(ScrollTo, mentioned in point 1 above) doesn't perform on a Salesforce application? However, the same works with a react-based web app.

**Reply by asolntsev** [reply, marked as answer]
hi.
> why the first method doesn't perform on a Salesforce application?

No, I don't know. Scrolling is not trivial. It heavily depends on your application. You probably need to use the second method `$.scrollIntoView(String)` and experiment with different options in this String parameter.

**Reply by aravindraveendran7** [reply]
Thank you @asolntsev .I tried with `scrollIntoView(string)` for some of scroll options after your reply.It did worked well.Was just curious to know how it behaves under the hood.Thanks a lot..
